shared/typeinference/codeql/typeinference/internal/TypeInference.qll

```diff
@@ -997,6 +1006,40 @@ module Make1<LocationSig Location, InputSig1<Location> Input1> {
         )
       }
 
+      /**
+       * Holds if `tt` does not satisfy `constraint`.
+       */
+      pragma[nomagic]
+      private predicate hasNotConstraintMention(HasTypeTree tt, Type constraint) {
```

Comment:
I find this implementation of this predicate hard to understand. I'm guessing it's best understood by seeing it as the negated (and then reduced) implementation of `hasConstraintMention`? Could we mention that in a comment? Or add some other comment to aid reading if there's a better way to understand it?